primitives/sidechain/primitives/src/types/block.rs

```diff
@@ -139,181 +139,47 @@ mod tests {
 	use sp_core::{ed25519, Pair};
-	use std::{
-		thread,
-		time::{Duration, SystemTime, UNIX_EPOCH},
-	};
-
-	/// sets the timestamp of the block as seconds since unix epoch
-	fn get_time() -> i64 {
-		SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs() as i64
-	}
-
-	#[test]
-	fn block_new_works() {
-		// given
-		let author: AccountId32 =
-			ed25519::Pair::from_string("//Alice", None).unwrap().public().into();
-		let block_number: u64 = 0;
-		let parent_hash = H256::random();
-		let layer_one_head = H256::random();
-		let signed_top_hashes = vec![];
-		let encrypted_payload: Vec<u8> = vec![];
-		let shard = ShardIdentifier::default();
-
-		// when
-		let block = Block::new(
-			author.clone(),
-			block_number,
-			parent_hash.clone(),
-			layer_one_head.clone(),
-			shard.clone(),
-			signed_top_hashes.clone(),
-			encrypted_payload.clone(),
-			get_time(),
-		);
-
-		// then
-		assert_eq!(block_number, block.block_number());
-		assert_eq!(parent_hash, block.parent_hash());
-		assert_eq!(layer_one_head, block.layer_one_head());
-		assert_eq!(shard, block.shard_id());
-		assert_eq!(&author, block.block_author());
-		assert_eq!(signed_top_hashes, *block.signed_top_hashes());
-		assert_eq!(encrypted_payload, *block.state_payload());
+	use std::time::{SystemTime, UNIX_EPOCH};
+
+	/// gets the timestamp of the block as seconds since unix epoch
+	fn timestamp_now() -> u64 {
+		SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_millis() as u64
+	}
+
+	fn test_block() -> Block {
+		Block::new(
+			ed25519::Pair::from_string("//Alice", None).unwrap().public().into(),
+			0,
+			H256::random(),
+			H256::random(),
+			H256::random(),
+			Default::default(),
+			Default::default(),
+			timestamp_now(),
+		)
 	}
 
 	#[test]
 	fn signing_works() {
-		// given
-		let signer_pair = ed25519::Pair::from_string("//Alice", None).unwrap();
-		let author: AccountId32 = signer_pair.public().into();
-		let block_number: u64 = 0;
-		let parent_hash = H256::random();
-		let layer_one_head = H256::random();
-		let signed_top_hashes = vec![];
-		let encrypted_payload: Vec<u8> = vec![];
-		let shard = ShardIdentifier::default();
-
-		// when
-		let block = Block::new(
-			author,
-			block_number,
-			parent_hash.clone(),
-			layer_one_head.clone(),
-			shard.clone(),
-			signed_top_hashes.clone(),
-			encrypted_payload.clone(),
-			get_time(),
-		);
+		let block = test_block();
+		let signer = ed25519::Pair::from_string("//Alice", None).unwrap();
 
 		let signature: Signature =
-			Signature::Ed25519(signer_pair.sign(block.encode().as_slice().into()));
-		let signed_block: SignedBlock = block.clone().sign_block(&signer_pair);
+			Signature::Ed25519(signer.sign(block.encode().as_slice().into()));
+		let signed_block: SignedBlock = block.clone().sign_block(&signer);
 
-		// then
 		assert_eq!(signed_block.block(), &block);
 		assert_eq!(signed_block.signature(), &signature);
-	}
-
-	#[test]
-	fn verify_signature_works() {
```

Comment:
Because as far as I've learned to goal of unittesting it to test one single thing - e..g. one assert!() per test, if that makes sense. So ideally it the verify_signature test would even test a home-designed SignedBlock.

Reply:
Hmm, yes, you are right about only one thing per unit test, and I tend to get a little to lazy sometimes in this regard. 

However, here I think that the `asserts` test the same thing. Or in fact, I think that we actually don't need to test what the signature is, we only need to check that the signature can be verified. (other signature schemes, e.g., are not even deterministic, so we can't do that check.)

I think we don't need to test a 'home-designed' signed-block because we can still construct a block wrongly afterwards, it has to many degrees of freedom. I would even go so far as not 't offer a public constructor and only allow creation of a `SignedBlock` via the `SignBlock` trait.

Reply:
I agree, it would be an overkill in this situation. But thanks for sharing your opinion, helps me alot.
I approved, so feel free to merge

Reply:
You are very welcome. :)